core/config/project_settings.cpp

```diff
@@ -1813,12 +1813,10 @@ ProjectSettings::ProjectSettings() {
 	GLOBAL_DEF(PropertyInfo(Variant::FLOAT, "rendering/rendering_device/pipeline_cache/save_chunk_size_mb", PROPERTY_HINT_RANGE, "0.000001,64.0,0.001,or_greater"), 3.0);
 	GLOBAL_DEF(PropertyInfo(Variant::INT, "rendering/rendering_device/vulkan/max_descriptors_per_pool", PROPERTY_HINT_RANGE, "1,256,1,or_greater"), 64);
 
-	GLOBAL_DEF_RST("rendering/rendering_device/d3d12/max_resource_descriptors_per_frame", 16384);
-	custom_prop_info["rendering/rendering_device/d3d12/max_resource_descriptors_per_frame"] = PropertyInfo(Variant::INT, "rendering/rendering_device/d3d12/max_resource_descriptors_per_frame", PROPERTY_HINT_RANGE, "512,262144");
-	GLOBAL_DEF_RST("rendering/rendering_device/d3d12/max_sampler_descriptors_per_frame", 1024);
-	custom_prop_info["rendering/rendering_device/d3d12/max_sampler_descriptors_per_frame"] = PropertyInfo(Variant::INT, "rendering/rendering_device/d3d12/max_sampler_descriptors_per_frame", PROPERTY_HINT_RANGE, "256,2048");
-	GLOBAL_DEF_RST("rendering/rendering_device/d3d12/max_misc_descriptors_per_frame", 512);
-	custom_prop_info["rendering/rendering_device/d3d12/max_misc_descriptors_per_frame"] = PropertyInfo(Variant::INT, "rendering/rendering_device/d3d12/max_misc_descriptors_per_frame", PROPERTY_HINT_RANGE, "32,4096");
+	GLOBAL_DEF_RST("rendering/rendering_device/d3d12/max_resource_descriptors", 65536);
```

Comment:
These changes will break compatibility unfortunately. Probably not in a way that matters, but I am not 100% sure this is safe to do. 

@akien-mga Do you have a better sense of whether deleting project settings like this will break projects? Alternatively, do we have a way of deprecating project settings gracefully?

Reply:
In general, removing project setting should not break projects on its own.

But if this could be set to some value (e.g., higher limit) that would not work with new default, old property still can be read by compatibility code:

```c++
#ifndef DISABLE_DEPRECATED
	if (ProjectSettings::get_singleton()->has_setting("rendering/rendering_device/d3d12/max_resource_descriptors_per_frame")) {
		int64_t limit = GLOBAL_GET("rendering/rendering_device/d3d12/max_resource_descriptors_per_frame");
		// adjust curent limit and pring warning
	}
#endif
```

Reply:
The meaning of both values are very incompatible since one was designed around current frame usage and the new one covers global application allocation, so I think the right move here is indeed to remove it like Skyth did. This is one of those values you'd only increase if your project ran into issues (and it's a reasonable limit by default to not overallocate excessively) and it should be very apparent for the user if it's necessary.